### ISSUE TITLE
Integration Requests

### ISSUE THREAD

**Issue by mfix22** [dawnlabs/carbon#358]
- Command Line Tool (https://github.com/dawnlabs/carbon/issues/52, https://github.com/dawnlabs/carbon/issues/240)
- IFrame/Embeddable (https://github.com/dawnlabs/carbon/issues/179)
- Remark.js Integration (https://github.com/dawnlabs/carbon/issues/209)
- Carbon API (https://github.com/dawnlabs/carbon/issues/210)
- Chrome Extension (https://github.com/dawnlabs/carbon/issues/219)
- Gatsby Plugin (https://github.com/dawnlabs/carbon/issues/312)
- Google Slides Integration (https://github.com/dawnlabs/carbon/issues/353)

/poll "Command Line Tool" "IFrame/Embeddable" "Remark.js Integration" "Carbon API" "Chrome Extension" "Gatsby Plugin" "Google Slides Integration"

**Comment by GavinRay97** [comment]
@mfix22 If you'd like I could try to write the CLI up sometime this week. I'm not sure about the how much work the API would be, I think I can just call `getCarbonImage()` and pass in params as the options.

Let me know if either would be helpful :ok_hand:

**Comment by mfix22** [comment]
@GavinRay97 i was always imagining that the initial CLI version would simply open a file up in the browser and allow flags to be set for theme and such (to start just just opening a file would be cool). This way no API is needed besides passing the correct query params.

If/when we open an API, the CLI could then later be updated to use it directly, but we shouldn't let that block the CLI effort. 

Happy to accept a PR adding it to our community section of the README 😄 

Couple notes: 
- [ ] Has to work cross-platform (here is an example our teammate made awhile back of how to open links in browser for the most popular OS's https://github.com/briandennis/WebLauncher)
- [ ] Would love if it was a Node CLI so people can install with NPM

**Comment by kaicianflone** [comment]
An instagram mode with resizable text based on the square dimensions would be nice too. Let me know where else I can help. I will go ahead and knock off some testing/easy issues.

**Comment by mfix22** [comment]
@kaicianflone there is already a Square dimensions mode for Instagram (: look under "Advanced" in the settings

**Comment by kaicianflone** [comment]
On my computer it's only exporting as square rather than sizing as square. Like some of my pictures will have the edges cutoff and some of the code missing.

**Comment by mfix22** [comment]
Feature: Create Gist (#393 )

**Comment by mfix22** [comment]
> Just like Github gist, add support for GitLab snippets - @mittalyashu (#464)

**Comment by mittalyashu** [comment]
@mfix22 if you planning to add all the features one by one, how will you structure your repository, will it be a mono-repo or multi-repo?

**Comment by mfix22** [comment]
We don't plan on supporting _all_ of the integrations, just the most popular/useful ones. 
A mono-repo 😄

**Comment by mfix22** [comment]
@mixn right, most of these integrations can be created/maintained separate from our project! (:

**Comment by mittalyashu** [comment]
@mfix22 I have few insight how we can deal with this feature -> IFrame/Embedded

## IFrame

Allow other users to embed this whole component on their website _(more people will use and it will increase popularity of carbon)_

![image](https://user-images.githubusercontent.com/29014463/44202108-ae6f4a80-a168-11e8-84ff-52a6a6c7512e.png)

### Possible options:

1. Limitation the use of carbon component on third-party website.
2. Unlimited usage of carbon component for verified third-party websites _(verification can be done via github, gitlab or bitbucket)_

### Where it can be embed?

1. Websites

## Embed

The user can embed the carbon as an editable image, where the viewers will see as an a code snippet converted into beautiful code which is copy able and for the admin that code snippet settings can be configured and even the code snippet can be changed.

### Where it can be embed?

1. Google Slides
2. Websites _(All types of websites: Dynamic, Static, Blogs)_
3. [Medium](https://github.com/dawnlabs/carbon/issues/179#issuecomment-362763999)
4. [Stackoverflow](https://stackoverflow.com/questions/5449872/css-notlast-childafter-selector/51204589?noredirect=1#comment90627990_51204589)

**Comment by maxcell** [comment]
@mittalyashu What you propose is exactly what I have been thinking about for quite sometime after coming across Carbon. I'm not sure what it would involve but I think it could be a great step in a solid direction.

**Comment by mfix22** [comment]
Embed support was just released with `3.5.0`! Try pasting any Carbon URL into Medium, or click the "Copy Embed" button to copy the `iframe` you can use in your own site or blog.

**Comment by mfix22** [comment]
Slack Integration (#533)

**Comment by mittalyashu** [comment]
@mfix22 I have tried pasting the carbon url. But nothing is showing on the screen.

https://medium.com/@mittalyashu/omo-b71b2a7529e2

![chrome_2018-10-15_11-36-48](https://user-images.githubusercontent.com/29014463/46932759-bcf5b780-d06e-11e8-9785-8ebd9218722c.png)

**Comment by mfix22** [comment]
@mittalyashu I'm going to hide your comment, but just because I'd like you to open issues not related to new features as new issues 🙂 Please do so for your comment above

**Comment by Jeepeo** [comment]
Cant see any API integration ...

**Comment by mittalyashu** [comment]
> Not sure if we are allowed to create an issue for new syntax

But can some add a syntax of Atom one dark theme?

**Comment by mfix22** [comment]
> But can some add a syntax of Atom one dark theme?

@mittalyashu this thread is only for integration requests — to integrate Carbon with other services and to integrate other services into Carbon. 

As for you question about adding the "Atom One Dark" theme, we already have "One Dark". Is there something preventing you from using the "Create+" a theme feature to create one for yourself?

<img width="80%" alt="Screenshot 2019-07-29 10 27 14" src="https://user-images.githubusercontent.com/8397708/62068726-7cc18900-b1eb-11e9-9a51-2060b15a1490.png">

**Comment by guthlStarkware** [comment]
Hi, thanks a lot for the amazing job here.
Any integration to rst format? It would help a lot for documentations.
Thanks

**Comment by mfix22** [comment]
@guthlStarkware could you open a new issue? I'm not sure what rst format is.

**Comment by guthlStarkware** [comment]
Will do. rst is RestructedText used in Sphinx: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html
I guess my request could be simplified to outputting an HTML file containing all the snippet+CSS+JS. I'll open an issue

**Comment by mfix22** [comment]
@guthlStarkware no need. We can track this under #1042 (which is very similar)

**Comment by Swordword** [comment]
I want to  Integrate to Remark.js, but use `embed` it will create iframe for every code block. so, is there a better way to accomplish it ?

**Comment by mfix22** [comment]
> I want to Integrate to Remark.js, but use `embed` it will create iframe for every code block. so, is there a better way to accomplish it ?

Something like this: https://github.com/carbon-app/gatsby-remark-embed-carbon @Swordword 

Also please open a new issue if you want to discuss this any further. Thanks!

**Comment by Lonli-Lokli** [comment]
I see in in readme that it can be integrated with markdown sites right?

Is there nay integration with dev.to? It is similar to medium and I cannot make it work even in Medium

**Comment by neerajcodes** [comment]
Chrome extension is better integration.

**Comment by mfix22** [comment]
Integration request for Confluence macro: https://github.com/carbon-app/carbon/issues/1305
